planner/core/planbuilder.go

```diff
@@ -1956,18 +1972,34 @@ func (b *PlanBuilder) buildAnalyzeFullSamplingTask(
 	names []string,
 	tbl *ast.TableName,
 	version int,
-) ([]AnalyzeColumnsTask, error) {
+	persistOpts bool,
+	rsOptionsMap map[int64]V2AnalyzeOptions,
```

Comment:
If we propagate this map as an input-argument and the caller can guarantee it is initialized, then we don't need to add it into the output-argument list, which looks more go-style~

Reply:
Another choice is removing it from the input-arg list and then initializing it inside and returning it as an output-arg. (the advantage of this is then the caller doesn't need to initialize the map)

Reply:
Done